corehq/ex-submodules/casexml/apps/case/tests/util.py

```diff
@@ -231,16 +232,16 @@ def _delete_all(db, viewname):
                 pass
 
 
+@unit_testing_only
 def delete_all_cases():
-    assert settings.UNIT_TESTING
     _delete_all(CommCareCase.get_db(), 'case/get_lite')
 
 
+@unit_testing_only
 def delete_all_xforms():
-    assert settings.UNIT_TESTING
     _delete_all(XFormInstance.get_db(), 'couchforms/all_submissions_by_domain')
 
 
+@unit_testing_only
```

Comment:
It hasn't specifically been a problem, but similar things (like leaving a comment on a deprecated function saying "DON'T USE THIS CODE") have been, and we've found the only truly effective solution is to make something unusable—forcing the user to really think through why they want to use it and intentionally violate or change whatever boundary the other person was trying to set.